stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/LinkAttachmentView.kt

```diff
@@ -35,6 +38,22 @@ internal class LinkAttachmentView : FrameLayout {
             binding.labelTextView.text = label.capitalize()
         }
 
-        binding.mediaAttachmentView.showAttachment(attachment)
+        binding.mediaAttachmentView.showAttachment(attachment, shouldSetClickListeners = false)
+    }
+
+    fun setLinkPreviewClickListener(longClickTarget: View, linkPreviewClickListener: LinkPreviewClickListener) {
+        setOnClickListener {
+            previewUrl?.let { url ->
+                linkPreviewClickListener.onLinkPreviewClick(url)
+            }
+        }
+        setOnLongClickListener {
+            longClickTarget.performLongClick()
+            true
+        }
```

Comment:
Leaving this unset or having it set but returning `false` doesn't work, right? Would be nice to get rid of the `longClickTarget` solution, at least here 😄

Reply:
Correct, the same problem as we had with LongClick inside movement method. I think I'll expose separate method for setting long click target